src/lib/timing.c

```diff
@@ -39,53 +39,51 @@
 
 ZEND_EXTERN_MODULE_GLOBALS(xdebug)
 
-// Windows
-#if PHP_WIN32
 static uint64_t xdebug_get_nanotime_abs(xdebug_nanotime_context *nanotime_context)
 {
-	FILETIME filetime;
-
-	if (nanotime_context->win_precise_time_func != NULL) {
-		nanotime_context->win_precise_time_func(&filetime);
-		return ((((uint64_t)filetime.dwHighDateTime << 32) + (uint64_t)filetime.dwLowDateTime) - WIN_TICKS_SINCE_1601_JAN_1)
-			* WIN_NANOS_IN_TICK;
+#if PHP_WIN32
+	// Windows, win_precise_time_func is only available on Win 8 and later.
+	{
+		FILETIME filetime;
+
+		if (nanotime_context->win_precise_time_func != NULL) {
+			nanotime_context->win_precise_time_func(&filetime);
+			return ((((uint64_t)filetime.dwHighDateTime << 32) + (uint64_t)filetime.dwLowDateTime) - WIN_TICKS_SINCE_1601_JAN_1)
+				* WIN_NANOS_IN_TICK;
+		}
 	}
+#endif
 
-	return 0;
-}
-
-// Linux/Unix with clock_gettime
-#elif HAVE_CLOCK_GETTIME
-static uint64_t xdebug_get_nanotime_abs(xdebug_nanotime_context *nanotime_context)
-{
-   struct timespec ts;
+#if HAVE_CLOCK_GETTIME
+	// Linux/Unix with clock_gettime
+	{
+	   struct timespec ts;
 
 #ifdef CLOCK_MONOTONIC
-	if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
-		return (uint64_t)ts.tv_sec * NANOS_IN_SEC + (uint64_t)ts.tv_nsec;
-	}
-#else
-	if (clock_gettime(CLOCK_REALTIME, &ts) == 0) {
-		return (uint64_t)ts.tv_sec * NANOS_IN_SEC + (uint64_t)ts.tv_nsec;
+		if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
+			return (uint64_t)ts.tv_sec * NANOS_IN_SEC + (uint64_t)ts.tv_nsec;
+		}
+#endif
```

Comment:
I would drop this, based on http://btorpey.github.io/blog/2014/02/18/clock-sources-in-linux/

> Since gettimeofday() calls clock_gettime(CLOCK_REALTIME, )...

`CLOCK_REALTIME` is only an alias to `gettimeofday` (and with this functions there were not reported issues to be non-present)

Reply:
On Linux, but does that also hold true on other Unix-y platforms? As I'm not sure, and I doubt it hurts, I will leave it in.

Reply:
I think yes, if there were no reported issues with `clock_gettime`, I do not think this provide any advantage.